project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/settings/MinimizationPluginState.kt

```diff
@@ -5,35 +5,61 @@ import org.plan.research.minimization.plugin.model.FunctionLevelStage
 import org.plan.research.minimization.plugin.model.MinimizationStage
 import org.plan.research.minimization.plugin.model.state.*
 
-import com.intellij.openapi.components.BaseState
+import com.intellij.openapi.components.*
 import com.intellij.util.xmlb.annotations.Property
 import com.intellij.util.xmlb.annotations.XCollection
+import org.jetbrains.annotations.NonNls
 
-class MinimizationPluginState : BaseState() {
-    var currentCompilationStrategy by enum<CompilationStrategy>(CompilationStrategy.GRADLE_IDEA)
+@Service(Service.Level.PROJECT)
+@State(
+    name = "org.intellij.sdk.settings.AppSettings",
+    storages = [Storage("ProjectMinimizationSettings.xml")],
+)
+class MinimizationPluginState : PersistentStateComponent<MinimizationPluginState.State>, BaseState() {
```

Comment:
It's a mistake to change this class, because the main idea of
- `MinimizationPluginState` is a class for defining settings only, without any functionality like yours `MinimizationPluginState.State`.
- `MinimizationPluginSettings` is a service that provides an API to get/modify/observe/save this state like yours `MinimizationPluginState`.